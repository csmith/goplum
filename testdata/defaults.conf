defaults {
    alerts = "text"
	good_threshold = 3
	interval = 1m30s
}

check debug.random "override1" {
    interval = 2s
    alerts = "foo"
}

check debug.random "override2" {
    interval = 3s
    good_threshold = 5
    failing_threshold = 6
}

check debug.random "defaults" {}

alert debug.sysout "text" {}
alert debug.sysout "foo" {}
