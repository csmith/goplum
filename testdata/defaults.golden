{
  "Alerts": {},
  "Checks": [
    {
      "Name": "override1",
      "Type": "debug.random",
      "Config": {
        "Alerts": [
          "foo"
        ],
        "Interval": 2000000000,
        "GoodThreshold": 3,
        "FailingThreshold": 2
      },
      "Check": {
        "PercentGood": 0.5
      },
      "LastRun": "0001-01-01T00:00:00Z",
      "Settled": false,
      "State": "indeterminate"
    },
    {
      "Name": "override2",
      "Type": "debug.random",
      "Config": {
        "Alerts": [
          "text"
        ],
        "Interval": 3000000000,
        "GoodThreshold": 5,
        "FailingThreshold": 6
      },
      "Check": {
        "PercentGood": 0.5
      },
      "LastRun": "0001-01-01T00:00:00Z",
      "Settled": false,
      "State": "indeterminate"
    },
    {
      "Name": "defaults",
      "Type": "debug.random",
      "Config": {
        "Alerts": [
          "text"
        ],
        "Interval": 90000000000,
        "GoodThreshold": 3,
        "FailingThreshold": 2
      },
      "Check": {
        "PercentGood": 0.5
      },
      "LastRun": "0001-01-01T00:00:00Z",
      "Settled": false,
      "State": "indeterminate"
    }
  ]
}